rir/src/utils/configurations.h

```diff
@@ -0,0 +1,39 @@
+#ifndef RIR_CONFIGURATIONS_H
+#define RIR_CONFIGURATIONS_H
+
+#include "utils/INIReader.h"
+#include "compiler/translations/pir_translator.h"
+
+namespace rir {
+
+struct Optimization {
+    Optimization(pir::PirTranslator* translator, short order) : translator(translator), order(order) {}
+    ~Optimization() { delete translator; }
+    
+    pir::PirTranslator* translator;
+    short order;
+
+    bool operator < (const Optimization& anotherOptimization) const {
+        if (order == anotherOptimization.order) {
+            return anotherOptimization.translator->getName().compare("cleanup");
```

Comment:
`compare` behaves like c `strcmp`, see [here](http://www.cplusplus.com/reference/string/string/compare/).
Therefore if `a` is shorter than `b`, this will return `-1`, which is `true` (everything except `0` is).

you probably want to return `compare > 0`